packages/crypto/src/ed25519.rs

```diff
@@ -33,37 +30,14 @@ pub const EDDSA_PUBKEY_LEN: usize = 32;
 /// - public key: raw ED25519 public key (32 bytes).
 pub fn ed25519_verify(message: &[u8], signature: &[u8], public_key: &[u8]) -> CryptoResult<bool> {
     // Validation
-    if message.len() > MESSAGE_MAX_LEN {
-        return Err(CryptoError::msg_err(format!(
-            "too large: {}",
-            message.len()
-        )));
-    }
-    if signature.len() != EDDSA_SIGNATURE_LEN {
-        return Err(CryptoError::sig_err(format!(
-            "wrong / unsupported length: {}",
-            signature.len()
-        )));
-    }
-    let pubkey_len = public_key.len();
-    if pubkey_len == 0 {
-        return Err(CryptoError::pubkey_err("empty"));
-    }
-    if pubkey_len != EDDSA_PUBKEY_LEN {
-        return Err(CryptoError::pubkey_err(format!(
-            "wrong / unsupported length: {}",
-            pubkey_len,
-        )));
-    }
-    // Deserialization
-    let signature = ed25519::Signature::try_from(signature)
-        .map_err(|err| CryptoError::generic_err(err.to_string()))?;
-
-    let public_key = ed25519::VerificationKey::try_from(public_key)
-        .map_err(|err| CryptoError::generic_err(err.to_string()))?;
+    check_message_length(message)?;
+    let signature = read_signature(signature)?;
+    let pubkey = read_pubkey(public_key)?;
 
     // Verification
-    match public_key.verify(&signature, &message) {
+    match VerificationKey::try_from(pubkey)
+        .and_then(|vk| vk.verify(&Signature::from(signature), &message))
+    {
```

Comment:
Why did you change this?

I guess this is causing the change in `import.rs:1629` below.

Reply:
At this point we know pubkey is `[u8; 32]`, so all the obvious problems are detected. Now everything else results in an `ed25518_zebra::Error` anyways, which does not provide helpful data. It also ensures pubkeys that are wrong but fall on a curve point by accident are handled the same way as wrong pubkeys that don't fall on a curve point. We could create a new, explicit `PubkeyNotOnCurve` if we wanted, but I don't think it is needed.

Reply:
Thanks for the explanation. I imagined something of the like, but didn't want to analyze it in detail.